core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -158,22 +161,29 @@ public List<Delete> getDeletesInDeleteSet() {
     return new ArrayList<>(deleteSet.values());
   }
 
-  public Optional<TransactionResult> get(Key key) throws CrudException {
+  public Optional<TransactionResult> mergeResult(Key key, Optional<TransactionResult> result)
+      throws CrudException {
     if (deleteSet.containsKey(key)) {
       return Optional.empty();
-    } else if (readSet.containsKey(key)) {
-      if (writeSet.containsKey(key)) {
-        // merge the result in the read set and the put in the write set
-        return Optional.of(
-            new TransactionResult(
-                new MergedResult(readSet.get(key), writeSet.get(key), getTableMetadata(key))));
-      } else {
-        return readSet.get(key);
-      }
+    } else if (writeSet.containsKey(key)) {
+      // merge the result in the read set and the put in the write set
+      return Optional.of(
+          new TransactionResult(
+              new MergedResult(result, writeSet.get(key), getTableMetadata(key))));
+    } else {
+      return result;
     }
-    throw new IllegalArgumentException(
-        CoreError.CONSENSUS_COMMIT_GETTING_DATA_NEITHER_IN_READ_SET_NOR_DELETE_SET_NOT_ALLOWED
-            .buildMessage());
+  }
+
+  public Optional<TransactionResult> mergeResult(
+      Key key, Optional<TransactionResult> result, Set<Conjunction> conjunctions)
+      throws CrudException {
+    return mergeResult(key, result)
+        .filter(
+            r ->
+                conjunctions.isEmpty()
```

Comment:
In 67446f1, I added a comment and changed the checking order for better understanding.

Reply:
@jnmt Sorry, one more thing.
I need a clarification for `We need to apply conditions if it is a merged result`.
Can you add the reason like `We need to apply conditions if it is a merged result because ...`?

Reply:
@feeblefakie Does the revised comment in d951832 make it clear? Thanks.